File: com_punkwm_dock/src/services/Settings/Google.js
class Google extends Sender {
    static async SaveMaxEventsView(value){super.Emit("Google", "Save Max Events View", {"Max":value});}
    static async SaveCredentials(value){super.Emit("Google", "Save Credentials", {"Credentials":value});}
    static async addScope(value){super.Emit("Google", "Add Scope", {"Scope":value});}
    static async popScope(value){super.Emit("Google", "Pop Scope", {"Scope":value});}
    static Load(Events, Scopes, Credentials){
        Storage.setItem("Google-Max-Events", Events);
        GoogleMaxEvents.value = getGoogleMaxEvents();
        Storage.setItem("Google-Scopes", JSON.stringify(Scopes));
        viewGoogleScopes();
        LoadCredentials(Credentials);
    }
    
}

File: com_punkwm_dock/src-tauri/src/ipc/settings/handlers/google.rs
use crate::{ipc::message::IpcMessage, utils::file::LocalFile};
use crate::apphandle::get_api_ipc;
// use punkwm_dock_lib::print_in_tty;

pub fn handler_google(msg: IpcMessage) {
     match msg.name().as_str() {
        "Save Max Events View" =>{
            let api = get_api_ipc();
            let max: u16 = msg.data()["Max"].to_string().replace("\"", "").parse::<u16>().unwrap_or(5);
            api.with_config(|cfg| {
               cfg.google().set_events(max);
            });
        },
        "Add Scope" =>{
            let api = get_api_ipc();
            let scope = msg.data()["Scope"].to_string();
            api.with_config(|cfg| {
               cfg.google().add_scope(scope);
            });

        },
        "Pop Scope" =>{
            let api = get_api_ipc();
            let scope = msg.data()["Scope"].to_string();
            api.with_config(|cfg| {
               cfg.google().pop_scope(scope);
            });

        },
        "Save Credentials" =>{
            let api = get_api_ipc();
            let credentials = msg.data()["Credentials"].to_string();
            api.with_config(|cfg| {
                let path = LocalFile::save_json_text_file(format!("{}/Credentials",cfg.folder()), credentials);
                cfg.google().set_credentials(path);
            });
        },

        _ => {
            println!(
                "Nombre desconocido: [{}:{}]",
                msg.category(),
                msg.name()
            );
        }
     }
    
}